### PR TITLE
Added validation to profile window, revamped it and user settings window

### PR DESCRIPTION
**Pull request by AndersMariegaard**
Added a few validation rules that can be used on any window for input validation.
Set user settings and profile windows to always be on top of main window.
Added a control template to display input validation errors above the control generating them (works for textboxes atleast)

### CODE REVIEW

File: DashboardFrontend/DetachedWindows/NewProfileWindow.xaml [hunk 1]
```diff
@@ -3,102 +3,199 @@
...
+                <ColumnDefinition Width="15"/>
+            </Grid.ColumnDefinitions>
+            <Grid.RowDefinitions>
+                <RowDefinition Height="30"/>
+                <RowDefinition Height="*"/>
+                <RowDefinition Height="42"/>
+            </Grid.RowDefinitions>
+
+            <Grid MouseLeftButtonDown="DraggableGrid"
+                  Grid.ColumnSpan="3">
```
Comment (mirakst) [inline DashboardFrontend/DetachedWindows/NewProfileWindow.xaml]:
```suggestion
```

File: DashboardFrontend/DetachedWindows/SettingsWindow.xaml.cs [hunk 2]
```diff
@@ -84,5 +101,20 @@ private void Button_SetActiveProfile(object sender, RoutedEventArgs e)
...
+
+        private void CommandBinding_Executed_1(object sender, ExecutedRoutedEventArgs e)
+        {
+            SystemCommands.CloseWindow(this);
+        }
+
+        private void DraggableGrid(object sender, MouseButtonEventArgs e)
+        {
+            this.DragMove();
+        }
```
Comment (mirakst) [inline DashboardFrontend/DetachedWindows/SettingsWindow.xaml.cs]:
```suggestion
```

**Review by mirakst — CHANGES_REQUESTED** [review]
fix plz

File: DashboardFrontend/DetachedWindows/NewProfileWindow.xaml.cs [hunk 3]
```diff
@@ -39,30 +40,61 @@ private void Button_Save(object sender, RoutedEventArgs e)
...
+
+        private void CommandBinding_Executed_1(object sender, ExecutedRoutedEventArgs e)
+        {
+            SystemCommands.CloseWindow(this);
+        }
+
+        private void DraggableGrid(object sender, MouseButtonEventArgs e)
+        {
+            this.DragMove();
         }
```
Comment (mirakst) [inline DashboardFrontend/DetachedWindows/NewProfileWindow.xaml.cs]:
```suggestion
```

File: DashboardFrontend/DetachedWindows/SettingsWindow.xaml [hunk 4]
```diff
@@ -4,292 +4,369 @@
...
+    <Border BorderBrush="{DynamicResource BorderColor}"
+            BorderThickness="1">
+        <Grid Background="{DynamicResource CanvasColor}">
+            <Grid.RowDefinitions>
+                <RowDefinition Height="48"/>
+                <RowDefinition Height="*"/>
+                <RowDefinition Height="48"/>
+            </Grid.RowDefinitions>
+
+            <Grid MouseLeftButtonDown="DraggableGrid">
```
Comment (mirakst) [inline DashboardFrontend/DetachedWindows/SettingsWindow.xaml]:
```suggestion
```

**Review by mirakst — CHANGES_REQUESTED** [review]

**Review by TorbjoernH — APPROVED** [review]

**Pull request merged**